src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,16 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * Use to detect mutating commands.
+     */
+    public Boolean isMutating() {
```

Comment:
Think about why this can be a pitfall. If you have more classes that need to use `isMutating`, then you have to keep adding `instanceof Class`